src/ragger/backend/speculos.py

```diff
@@ -246,6 +247,29 @@ def compare_screen_with_text(self, text: str) -> bool:
                 return True
         return False
 
+    def _wait_for_text_on_screen_or_not(self,
+                                        should_be_on_screen: bool,
+                                        text: str,
+                                        timeout: float = 10.0) -> None:
+        endtime = time() + timeout
+        # Only manual ticks sent by compare_screen_with_text in this function because
+        # we don't want a desync between screen and events
+        self.pause_ticker()
+        while True:
+            if self.compare_screen_with_text(text) == should_be_on_screen:
+                # TODO: investigate while this line is needed.. It should not be
+                self._last_screenshot = BytesIO(self._client.get_screenshot())
```

Comment:
Did you try putting it betwwen the `pause_ticker` and the `while True`?

Reply:
Yep, does not work
also tried putting it between the `while True` and the `self.compare_screen_with_text`, does not work either